src/plugins/intel_cpu/src/nodes/reduce.cpp

```diff
@@ -910,7 +918,7 @@ struct jit_uni_reduce_kernel_f32 : public jit_uni_reduce_kernel, public jit_gene
                 assert(!"unknown src_dt");
         }
 
-        if (!isFloatCompatible(src_dt))
+        if (!isFloatCompatible(src_dt) && jcp_.reduce_mode != Algorithm::ReduceProd)
             uni_vcvtdq2ps(vmm_src, vmm_src);
```

Comment:
Instead of doing such an explicit check, can we replace this by adding a flag, something like:
`jcp_.convert_int`, So that when we further extend the reduce modes with native int32 support, we simply extend the dedicated check for setting this flag, instead of modifying the `if` condition in several places of code? And it is even better to replace the `!isFloatCompatible(src_dt)` with more common check, like 
```cpp
bool convert_int(data_type) {
    return !isFloatCompatible(src_dt) && jcp_.convert_int;
    //here can be other conditions as well
}
```

```cpp
//kernel configuration stage
jcp_.convert_int = !(one_of(jcp_.reduce_mode, Algorithm::ReduceProd, /*may be some other modes that support native integer arithmetic in future*/);
....
inline void load_vector(Vmm vmm_src, const Xbyak::Address &op, memory::data_type src_dt) {
...
if (convert_int(src_dt)) {
    uni_vcvtdq2ps(vmm_src, vmm_src);
...
}
```
What do you think?

Reply:
Good point! Applied. And as reduce mode(Algorithm type) has already been passed into the JIT kernel by jit_reduce_config_params, I just calculated the flag inside JIT kernel. Please feel free to have further discussiones, if you have concerns about that. Thanks Maksim!